propagation/propagation.go

```diff
@@ -29,6 +29,27 @@ type TextMapCarrier interface {
 	// must never be done outside of a new major release.
 }
 
+// MultiTextMapCarrier is a TextMapCarrier that can return multiple values for a single key.
+type MultiTextMapCarrier interface {
+	// DO NOT CHANGE: any modification will not be backwards compatible and
+	// must never be done outside of a new major release.
+
+	TextMapCarrier
+	MultiGetter
+}
+
+// MultiGetter can return multiple values for a single key,
+// with contrast to TextMapCarrier.Get which returns a single value.
+type MultiGetter interface {
```

Comment:
Shouldn't we call out in `TextMapPropagator.Extract` documentation that implementations should check if the carrier implements this interface and use it to support multiple values?

Reply:
```
	// Implementations may check if the carrier implements MultiTextMapCarrier,
	// to support usage of multiple values per key.
```

I've implemented your suggestion, but used the word "may" instead of "should" since not all propagators will necessarily want to support multi-value (for example, I imagine B3 would not).